src/controller/tripController.go

```diff
@@ -2,40 +2,397 @@ package controller
+	getUserIdQueryString := "SELECT id FROM \"user\" WHERE username = $1"
+	row = tc.DatabaseMgr.ExecuteQueryRow(getUserIdQueryString, inviteUserRequest.Username)
+	var userId uuid.UUID
+	if err := row.Scan(&userId); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, expenseerror.EXPENSE_USER_NOT_FOUND
+		}
+		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	// Check if tokenUser is part of trip
+	var count int
+	checkUserTripQueryString := "SELECT COUNT(*) FROM user_trip_association WHERE id_trip = $1 AND id_user = $2"
+	row = tc.DatabaseMgr.ExecuteQueryRow(checkUserTripQueryString, tripId, tokenUserId)
+	if err := row.Scan(&count); err != nil {
+		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	if count == 0 {
+		return nil, expenseerror.EXPENSE_FORBIDDEN
+	}
+
+	// Add user to trip
+	addUserToTripQueryString := "INSERT INTO user_trip_association (id_trip, id_user, is_accepted) VALUES ($1, $2, $3)"
+	if _, err := tc.DatabaseMgr.ExecuteStatement(addUserToTripQueryString, tripId, userId, false); err != nil {
+		// if err is unique_violation, then user is already part of trip
+		if pqErr, ok := err.(*pq.Error); ok {
+			if pqErr.Code == "23505" {
+				return nil, expenseerror.EXPENSE_CONFLICT
+			}
+		}
+		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteStatement(): %v", err)
+		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+	return tripDetails, nil
+}
+
+func (tc *TripController) AcceptTripInvite(ctx context.Context, tripId *uuid.UUID) *model.ExpenseServiceError {
+	if utils.ContainsEmptyString(tripId.String()) {
+		return expenseerror.EXPENSE_BAD_REQUEST
+	}
+
+	// Get trip details
+	tripDetails := &model.TripResponse{}
+	getTripDetailsQueryString := "SELECT id, location, start_date, end_date FROM trip WHERE id = $1"
+	row := tc.DatabaseMgr.ExecuteQueryRow(getTripDetailsQueryString, tripId)
+	if err := row.Scan(&tripDetails.TripID, &tripDetails.Location, &tripDetails.StartDate, &tripDetails.EndDate); err != nil {
+		if sql.ErrNoRows == err {
+			return expenseerror.EXPENSE_TRIP_NOT_FOUND
+		}
+		log.Printf("Error in tripController.AcceptTripInvite.DatabaseMgr.ExecuteQueryRow(): %v", err)
+		return expenseerror.EXPENSE_UPSTREAM_ERROR
+	}
+
+	// Get authenticated user id from context
+	tokenUserId, ok := ctx.Value(model.ExpenseContextKeyUserID).(*uuid.UUID)
+	if !ok {
+		log.Printf("Error in tripController.AcceptTripInvite.ctx.Value(): %v", ok)
+		return expenseerror.EXPENSE_INTERNAL_ERROR
+	}
+
+	// Update user_trip_association
+	updateUserTripQueryString := "UPDATE user_trip_association SET is_accepted = true WHERE id_trip = $1 AND id_user = $2"
+	var result sql.Result
+	var err error
+	if result, err = tc.DatabaseMgr.ExecuteStatement(updateUserTripQueryString, tripId, tokenUserId); err != nil {
```

Comment:
```suggestion
	var result sql.Result
	if result, err = tc.DatabaseMgr.ExecuteStatement(updateUserTripQueryString, tripId, tokenUserId); err != nil {
```

Reply:
`err` wurde im scope drüber nicht davor definiert. Es muss zuerst Deklariert werden.

Reply:
Hab meine Suggestion angepasst, sollte kein Problem sein mit `if result, err := ...`

Reply:
`result` muss davor deklariert werden, damits außerhalb vom if-scope genutzt werden kann

Reply:
`# github.com/Travel-Utilities-WWI21SEB/expense-management-service/src/controller
src/controller/tripController.go:379:5: result declared and not used`

```suggestion
	var result sql.Result
	if result, err := tc.DatabaseMgr.ExecuteStatement(updateUserTripQueryString, tripId, tokenUserId); err != nil {
		log.Printf("Error in tripController.AcceptTripInvite.DatabaseMgr.ExecuteStatement(): %v", err)
		return expenseerror.EXPENSE_UPSTREAM_ERROR
	}
```

geht nicht

Reply:
Dann so: 

```suggestion
	result, err = tc.DatabaseMgr.ExecuteStatement(updateUserTripQueryString, tripId, tokenUserId)
	if err != nil {
```

Reply:
Das Problem ist wirklich nur, dass du das `result` im scope oben drüber wieder verwenden möchtest:
`src/controller/tripController.go:378:5: result declared and not used
src/controller/tripController.go:384:23: undefined: result
`

Weil später willst du ja result.RowsAffected() aufrufen, was ja über den Scope vom if-Statement ist